From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 12 Nov 2017 01:03:55 +0000
Subject: swap: Avoid ABI change in 4.13.12
Forwarded: not-neede

Commit 2628bd6fc052 "mm, swap: fix race between swap count
continuation operations" added a field to struct swap_info_struct.
swap_info_struct is always allocated in built-in code and modules
won't use this new field, so move it to the end of the structure and
hide it from genksyms.

---
--- a/include/linux/swap.h
+++ b/include/linux/swap.h
@@ -246,12 +246,14 @@ struct swap_info_struct {
 					 * both locks need hold, hold swap_lock
 					 * first.
 					 */
+	struct work_struct discard_work; /* discard worker */
+	struct swap_cluster_list discard_clusters; /* discard clusters list */
+#ifndef __GENKSYMS__
 	spinlock_t cont_lock;		/*
 					 * protect swap count continuation page
 					 * list.
 					 */
-	struct work_struct discard_work; /* discard worker */
-	struct swap_cluster_list discard_clusters; /* discard clusters list */
+#endif
 };
 
 /* linux/mm/workingset.c */
